app/models/spree/license_key.rb

```diff
@@ -7,6 +7,15 @@ class LicenseKey < ActiveRecord::Base
     attr_accessible :license_key, :inventory_unit_id, :variant_id, :void, :memo, :activated_on
 
     scope :available, where(inventory_unit_id: nil, void: false)
-    scope :used, where('(inventory_unit_id IS NOT NULL) OR (void = ?)', true)
+    scope :used, where('inventory_unit_id IS NOT NULL')
+    scope :void, where(void: true)
+
+    def used?
+      self.class.used.where(id: id).any?
```

Comment:
I think the duplication is not a problem. re-querying for `self` is more problematic.